server/tests/src/test/java/org/infinispan/server/rollingupgrade/DefaultRollingUpgradeTest.java

```diff
@@ -10,4 +21,48 @@ public void testDefaultSetting() throws InterruptedException {
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder("15.2.0.Final", "15.2.1.Final");
       RollingUpgradeHandler.performUpgrade(builder.build());
    }
+
+   @Test
+   public void testHttpClient() throws Exception {
+      String cacheName = "rolling-upgrade";
+      TestUser user = TestUser.ADMIN;
+      RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder("15.2.0.Final", "15.2.1.Final")
+            .nodeCount(3);
+      RestClientConfigurationBuilder restBuilder = new RestClientConfigurationBuilder();
+      restBuilder.security().authentication().enable().username(user.getUser()).password(user.getPassword());
+      AtomicBoolean clientUtilized = new AtomicBoolean(false);
+      ByRef.Integer node = new ByRef.Integer(0);
+
+      builder.handlers(
+            uh -> {
+               // First creates the cache as replicated.
+               uh.getRemoteCacheManager()
+                     .administration()
+                     .createCache(cacheName, new StringConfiguration("<replicated-cache></replicated-cache>"));
+
+               // Since this is the first check, we can connect to any of the nodes in the previous version.
+               RestResponse res = await(uh.restFrom(0, restBuilder).cache(cacheName).put("foo", "bar"));
```

Comment:
This is fine for now, but we will want to change this to hit the liveness endpoint instead. But we can do that after.. as we will want basic put/get tested by REST as well.

Reply:
Done a small update here to use the health API. I am validating only for HTTP 200, I am not checking the JSON contents.